src/perfcollect/perfcollect

```diff
@@ -1280,53 +1304,61 @@ CreateLTTngSession()
 
 SetupLTTngSession()
 {
-    # Setup per-event context information.
-    RunSilent "lttng add-context --userspace --type vpid"
-    RunSilent "lttng add-context --userspace --type vtid"
-    RunSilent "lttng add-context --userspace --type procname"
-
-    if [ "$action" == "livetrace" ]
-    then
-        RunSilent "lttng enable-event --userspace --tracepoint DotNETRuntime:EventSource"
-    elif [ "$gcCollectOnly" == "1" ]
-    then
-        usePerf=0
-        EnableLTTngEvents ${DotNETRuntime_GCKeyword_GCCollectOnly[@]}
-        EnableLTTngEvents ${DotNETRuntimePrivate_GCPrivateKeyword_GCCollectOnly[@]}
-        EnableLTTngEvents ${DotNETRuntime_ExceptionKeyword[@]}
-    elif [ "$gcOnly" == "1" ]
+    # Setup kernel events. 
+    # TODO: enable triggers of different kernel events, instead of a default set of kernel events
+    if [ "$events" == "kernel" ]
     then
-        usePerf=0
-        EnableLTTngEvents ${DotNETRuntime_GCKeyword[@]}
-        EnableLTTngEvents ${DotNETRuntimePrivate_GCPrivateKeyword[@]}
-        EnableLTTngEvents ${DotNETRuntime_JitKeyword[@]}
-        EnableLTTngEvents ${DotNETRuntime_LoaderKeyword[@]}
-        EnableLTTngEvents ${DotNETRuntime_ExceptionKeyword[@]}
-    elif [ "$gcWithHeap" == "1" ]
-    then
-        usePerf=0
-        EnableLTTngEvents ${DotNETRuntime_GCKeyword[@]}
-        EnableLTTngEvents ${DotNETRuntime_GCHeapSurvivalAndMovementKeyword[@]}
+        RunSilent "lttng add-context --kernel -t pid -t procname"
```

Comment:
Rather than treating the kernel events differently in SetupLTTngSession, it would be great to have them interleaved into the rest of the code.  It is safe to run add-context even if no events come through.  So you should be OK to just add context to both the kernel and user space channels unconditionally.  The one question is whether or not it's safe to call lttng add-context --kernel if there is no kernel space channel (e.g. if the LTTng kernel modules aren't loaded).  Can you check this?

Then, you should just be able to add another if statement on the same level as if [ "$events" == "" ] and just call EnableLTTngKernelEvents.

Reply:
Good question, could you clarify  " (e.g. if the LTTng kernel modules aren't loaded). " ? I assume we always add context before channel is created? 
Based on the original code, we add context first, and the default channel will be created when we enable the events.